common/archiver/archivalMetadata.go

```diff
@@ -148,34 +123,53 @@ func NewArchivalConfig(
 	}
 
 	return &archivalConfig{
-		clusterStatus:       clusterStatus,
-		enableRead:          enableRead,
-		domainDefaultStatus: domainDefaultStatus,
-		domainDefaultURI:    domainDefaultURI,
+		staticClusterStatus:  staticClusterStatus,
+		dynamicClusterStatus: dc.GetStringProperty(dynamicconfig.HistoryArchivalStatus, staticClusterStatusStr),
```

Comment:
NewArchivalConfig seems like it is meant to work for both visibility and history but as its written this won't work for both?

Reply:
Makes sense. I've updated the function to take dynamic function as parameter.